src/medley/core.cljc

```diff
@@ -357,6 +357,35 @@
   ([pred coll]
    (rest (drop-while (complement pred) coll))))
 
+(defn partition-upto
+  "Returns a lazy sequence of partitions, splitting after `(pred item)` returns
+  true. Returns a transducer when no collection is provided."
+  ([pred]
+   (fn [rf]
+     (let [v (volatile! [])
+           split? (volatile! false)]
+       (fn
+         ([] (rf))
+         ([result]
+          (let [result (if-some [v (not-empty @v)]
+                         (unreduced (rf result v))
+                         result)]
+            (rf result)))
```

Comment:
Perhaps remove the `let` so that it's:

```clojure
          (rf (if-some [v (not-empty @v)]
                (unreduced (rf result v))
                result))
```

Reply:
Good spot, thanks